src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/4.sql

```diff
@@ -1629,6 +1702,59 @@ AS
     COMMIT TRANSACTION
 GO
 
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     AddExtendedQueryTagError
+--
+-- DESCRIPTION
+--    Adds an Extended Query Tag Error or Updates it if exists.
+--
+-- PARAMETERS
+--     @tagKey
+--         * The related extended query tag's key
+--     @errorCode
+--         * The error code
+--     @watermark
+--         * The watermark
+--     @createdTime
+--         * The time the error was created
+--
+-- RETURN VALUE
+--     The tag key of the error added.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.AddExtendedQueryTagError (
+    @tagKey INT,
+    @errorCode INT,
+    @watermark BIGINT,
+    @createdTime DATETIME2(7)
+)
+AS
+    SET NOCOUNT     ON
+    SET XACT_ABORT  ON
+    BEGIN TRANSACTION
+
+        --Check if instance exists
+        IF NOT EXISTS (SELECT * FROM dbo.Instance WITH (HOLDLOCK) WHERE Watermark = @watermark AND Status = 1)
```

Comment:
If all the queries are holding lock to support no parallel deletion, then UPDLOCK is enough.
I don't see any insertions for the conditions